File: 2_6/2_6.go
package main

import (
	"fmt"
)

func countNum(nums []int) int {
	result := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] >= 100 {
			count := 0
			tempNum := nums[i]
			for tempNum != 0 {
				tempNum /= 10
				count++
			}
			if count%2 != 0 {
				result++
			}
		}
	}
	return result
}

func main() {
	var nums []int
	fmt.Println("Enter a list of numbers (to end enter -1):")
	for {
		var num int
		fmt.Scan(&num)
		if num == -1 {
			break
		}
		nums = append(nums, num)
	}
	fmt.Println(countNum(nums))
}

File: 2_6/2_6.cpp
#include <iostream>
#include <vector>
using namespace std;

int countNum(vector<int> nums){
    int result = 0;
    for (int num : nums){
        int count = 0;
        if (num >= 100){ // кол во чисел 
        while (num != 0){
            num /= 10;
            count++;
        }
        if (count % 2 != 0){
            result++;
        }
    }
    }
    return result;
}


int  main(){
    vector<int> nums;
    int num;
    cout << "Enter a list of numbers (to end enter -1):" << endl;
    while (cin >> num && num != -1) { // пока ввод соответствует num ( тоесть это цифра ) и не введено -1
        nums.push_back(num);
    }
    cout  << countNum(nums) << endl;
    return 0;
}